### PR TITLE
fix(k8s): fix `flush and reshard` nemesis for K8S

### PR DESCRIPTION
**Pull request by vponomaryov**
Recently merged PR (https://github.com/scylladb/scylla-cluster-tests/pull/6773) add support to the multiple K8S clusters and updated the way how the `flush and reshard` nemesis works on K8S. For now it broken with the following error:

```
  Traceback (most recent call last):
  File "%path%/sdcm/nemesis.py", line 5069, in wrapper
    result = method(*args[1:], **kwargs)
  File "%path%/sdcm/nemesis.py", line 1381, in disrupt_nodetool_flush_and_reshard_on_kubernetes
    self._verify_resharding_on_k8s(current_cpus, dc_idx)
  File "%path%/sdcm/nemesis.py", line 1293, in _verify_resharding_on_k8s
    for node in reversed(qualified_nodes):
  TypeError: 'generator' object is not reversible
```

So, fix this typo.
Also, reduce the required number of needed Scylla cores to be 7 to cover more instance types such as `i3.2xlarge` and `i4i.2xlarge`.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
Tested here: https://jenkins.scylladb.com/job/scylla-staging/job/valerii/job/vp-longevity-scylla-operator-3h-eks/83/consoleFull

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -1357,7 +1356,7 @@ def disrupt_nodetool_flush_and_reshard_on_kubernetes(self):
             raise UnsupportedNemesis('It is supported only on kubernetes')
         dc_idx = 0
         for node in self.cluster.nodes:
-            if hasattr(node.k8s_cluster, 'eks_cluster_version') and node.scylla_shards >= 14:
+            if hasattr(node.k8s_cluster, 'eks_cluster_version') and node.scylla_shards >= 7:
```
Comment (fruch) [inline sdcm/nemesis.py:1359]: You might want to update the comments below

**Review by fruch — APPROVED** [review]
Lgtm

Smal nitpick

File: sdcm/nemesis.py (hunk 1 above)
Comment (vponomaryov) [reply sdcm/nemesis.py:1359]: Done

**Pull request merged**